component/src/main/java/org/wso2/extension/siddhi/io/googlepubsub/sink/GooglePubSubSink.java

```diff
@@ -0,0 +1,219 @@
+                        + "provided topic.id and project.id. Then the messages are published to the particular topic.",
+
+                        syntax = "@sink(type = 'googlepubsub', @map(type= 'text'),\n"
+                                + "project.id = 'sp-path-1547649404768', \n"
+                                + "credential.path = 'src/test/resources/security/sp.json',\n"
+                                + "topic.id ='topicA',\n "
+                                + ")\n" +
+                                "define stream inputStream(message string);"),
+        }
+)
+/*
+ * GooglePubSub publishing tasks.
+ */
+public class GooglePubSubSink extends Sink {
+
+    private static final Logger log = Logger.getLogger(GooglePubSubSink.class);
+    private String streamID;
+    private String siddhiAppName;
+    private GoogleCredentials credentials;
+    private TopicAdminClient topicAdminClient;
+    private String projectId;
+    private ProjectTopicName topic;
+    private Publisher publisher;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+
+        return new Class[]{String.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+
+        return new String[]{GooglePubSubConstants.TOPIC_ID, GooglePubSubConstants.GOOGLE_PUB_SUB_SERVER_PROJECT_ID,
+                GooglePubSubConstants.CREDENTIAL_PATH};
+    }
+
+    @Override
+    protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+                        SiddhiAppContext siddhiAppContext) {
+
+        this.siddhiAppName = siddhiAppContext.getName();
+        this.streamID = streamDefinition.getId();
+        String topicId = optionHolder.validateAndGetStaticValue(GooglePubSubConstants.TOPIC_ID);
+        this.projectId = optionHolder.validateAndGetStaticValue(GooglePubSubConstants.GOOGLE_PUB_SUB_SERVER_PROJECT_ID);
+        String credentialPath = optionHolder.validateAndGetStaticValue(GooglePubSubConstants.CREDENTIAL_PATH);
+        this.topic = ProjectTopicName.of(projectId, topicId);
+        File credentialsPath = new File(credentialPath);
+        try {
+            FileInputStream serviceAccountStream = new FileInputStream(credentialsPath);
+            credentials = ServiceAccountCredentials.fromStream(serviceAccountStream);
+        } catch (IOException e) {
+            throw new SiddhiAppCreationException("The file that contains your service account credentials is not "
+                    + "found or you are not permitted to make authenticated calls. Check the credential.path '"
+                    + credentialPath + "' defined in stream " + siddhiAppName + " : " + streamID + ".", e);
+        }
+        createTopic();
+        try {
+            publisher = Publisher.newBuilder(topic).setCredentialsProvider(FixedCredentialsProvider.create(credentials))
```

Comment:
Does it make sense to retry this request assuming failure occurred due to network/connectivity issue. If so we need to move this block to connect()